lib/activerecord/delay_touching/state.rb

```diff
@@ -49,6 +49,13 @@ def clear_records
         @already_updated_records.clear
       end
 
+      def remove_unpersisted_records!
+        @records.each do |_, set|
+          set.instance_variable_get(:@hash).rehash
```

Comment:
I think this may rely on ActiveRecord's `sync_with_transaction_state` being invoked upstream. From the AR source, that method is lazily executed, and if it's not run, I think rehash will have no effect.

For that reason, I think it's clearer to invoke explicitly here (as suggested in #14), whether it's via the private method itself or some public API.

Reply:
Then I'm surprised this works? Hmm.